tests/delete/tests.py

```diff
@@ -90,6 +90,30 @@ def test_protect_multiple(self):
         with self.assertRaisesMessage(ProtectedError, msg):
             a.protect.delete()
 
+    def test_protect_chained(self):
+        p = P.objects.create()
+        r = R.objects.create(p=p)
+        B.objects.create(protect=r)
+
+        msg = (
+            "Cannot delete some instances of model 'P' because they are "
+            "referenced through protected foreign keys: 'B.p'."
+        )
+        with self.assertRaisesMessage(ProtectedError, msg):
+            p.delete()
+
+    def test_restrict_chained(self):
```

Comment:
I think we don't need this test

Reply:
Yes we can remove it, it's redundant with `test_restrict_path_cascade_indirect()` and `test_restrict_path_cascade_direct()`, i.e. we have tests for a path "restricted" deletion.